### PR TITLE
feat(layout): Add cx prop, pseudo props, and other style props to Box

### PR DESCRIPTION
**Pull request by alanbsmith**
## Summary

category

---

Things still to be done:

- [ ] Clean up code
- [ ] Improve / clean up the types
  - [ ] Probably update csstype to v3?
- [ ] Add JSDoc comments
- [ ] Fix and add tests
- [ ] Add storybook docs 
  - [ ] Move `modules/labs-react/common/stories/examples/PropTables.splitprops.tsx` into `react/common`

## Checklist

- [ ] tests are changed or added
- [ ] code has been documented
- [ ] design approved final implementation
- [ ] a11y approved final implementation
- [ ] code adheres to the [API & Pattern guidelines](https://workday.github.io/canvas-kit/?path=/story/welcome-dev-docs-api-pattern-guidelines--page)

### CODE REVIEW

File: modules/react/layout/index.ts [hunk 1]
```diff
@@ -12,5 +12,5 @@ export * from './lib/utils/position';
 export {SpaceStyleProps} from './lib/utils/space';
 export * from './lib/Flex';
 export * from './lib/Stack';
-export type { FlexStyleProps } from './lib/utils/flex';
-export type { StackSpacing, StackDirection, StackStyleProps } from './lib/utils/stack';
+export type {FlexStyleProps} from './lib/utils/flex';
+export type {StackSpacing, StackDirection, StackStyleProps} from './lib/utils/stack';
```
Comment (alanbsmith) [inline modules/react/layout/index.ts:16]: looks like my linter went after this file.

File: modules/react/layout/lib/utils/cx.ts [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+  FlexItemStyleProps &
+  LayoutStyleProps &
+  OtherStyleProps &
+  PositionStyleProps &
+  SpaceStyleProps &
+  PseudoStyleProps & {
+    [key: string]: string | number;
+  };
+
+export function cxFn<S extends CXStyleProps>(styles: S, isRTL: boolean) {
```
Comment (alanbsmith) [inline modules/react/layout/lib/utils/cx.ts:25]: This currently does not support nested style objects.

File: modules/react/layout/lib/Box.tsx [hunk 3]
```diff
@@ -1,7 +1,10 @@
+/** @jsxRuntime classic */
+/** @jsx jsx */
+import {jsx} from '@emotion/react';
```
Comment (alanbsmith) [inline modules/react/layout/lib/Box.tsx:3]: I'm not sure this is the way we want to go about this or not. I'm open to suggestions here. We could also try to make this a style function similar to the other style props.

**Comment by mannycarrera4** [comment]
i know in our yarn.lock file we have csstype3 but in our react/package.json we have it to 2.x, we can upgrade this since i think emotion uses csstype3

File: modules/react/layout/lib/Box.tsx (hunk 3 above)
Comment (mannycarrera4) [reply modules/react/layout/lib/Box.tsx:3]: I think when we were talking about this this is what would work, have the pragma in one file where we use the css prop and that's it, im ok with this!

**Comment by alanbsmith** [comment]
Going to close this until I have some real time to dedicate to it.

**Pull request closed without merging**